pages/api/savedInsulinData/index.js

```diff
@@ -0,0 +1,22 @@
+import dbConnect from '../../../dbConnect';
+import SavedInsulinData from '../../../models/InsulinApp';
+import { getAllCategories } from '../../../services/insulinAppDataService';
+
+export default async function handler(request, response) {
+  switch (request.method) {
+    case 'GET':
+      const savedInsulinData = await getAllCategories();
+      return response.status(200).json({ savedInsulinData });
+
+    case 'DELETE':
+      await dbConnect();
+
+      const data = request.body;
+
+      await SavedInsulinData.findOneAndDelete();
+
+      return response
+        .status(201)
+        .json({ message: 'SavedInsulinData was created' });
+  }
```

Comment:
You don't need the DELETE case here and could go with a simple `if...else`:

```suggestion
if(request.method === "GET"){
      const savedInsulinData = await getAllCategories();
      response.status(200).json({ savedInsulinData });
      } else {
      response.status(405).json({message: "Method not allowed"})
      };
```

Reply:
Thanks for the good feedback, I have changed a few things.